compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -154,10 +154,11 @@ object TypeApplications {
   }
 }
 
-import TypeApplications._
-
 /** A decorator that provides methods for modeling type application */
-class TypeApplications(val self: Type) extends AnyVal {
+trait TypeApplications extends Any {
+  def self: Type
```

Comment:
That will cause a box for every method call. So don't think this is an improvement.